### DISCUSSION TITLE
`return` is for early results in Motoko (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4831] [Show and tell]
You can prematurely leave a Motoko function (or `async` expression) with the `return ⟨res⟩` expressions, should the circumstances allow it. (Here `⟨res⟩` is optional and defaults to unit: `()`). Notably though, you don't need a `return` when falling off the end of the function with a result (as most modern languages, like Rust and Wasm). We suggest that you only early-`return`.

**Comment by jessiemongeon1** [comment]
Scheduled for Friday May 30
